modules/ducktests/tests/checks/utils/check_marks.py

```diff
@@ -0,0 +1,240 @@
+
+    def check_with_others_marks(self):  # pylint: disable=R0201
+        """
+        Checks that ignite version parametrization works with others correctly.
+        """
+        @ignite_versions(str(DEV_BRANCH), str(V_2_8_1), version_prefix='ver')
+        @parametrize(x=10, y=20)
+        @parametrize(x=30, y=40)
+        def function_parametrize(ver, x, y):
+            return ver, x, y
+
+        @ignite_versions((str(DEV_BRANCH), str(V_2_8_1)), (str(V_2_8_1), str(V_2_7_6)), version_prefix='pair')
+        @matrix(i=[10, 20], j=[30, 40])
+        def function_matrix(pair_1, pair_2, i, j):
+            return pair_1, pair_2, i, j
+
+        @ignore(ver=str(DEV_BRANCH))
+        @ignite_versions(str(DEV_BRANCH), str(V_2_8_1), version_prefix='ver')
+        def function_ignore(ver):
+            return ver
+
+        context_list = expand_function(func=function_parametrize, sess_ctx=mock_session_ctx())
+        context_list += expand_function(func=function_matrix, sess_ctx=mock_session_ctx())
+        context_list += expand_function(func=function_ignore, sess_ctx=mock_session_ctx())
+
+        assert len(context_list) == 14
+
+        parametrized_context = list(filter(lambda x: x.function_name == function_parametrize.__name__, context_list))
+        assert len(parametrized_context) == 4
+        for ctx in parametrized_context:
+            args = ctx.injected_args
+            assert len(args) == 3
+            assert ctx.function() == (args['ver'], args['x'], args['y'])
+
+        matrix_context = list(filter(lambda x: x.function_name == function_matrix.__name__, context_list))
+        assert len(matrix_context) == 8
+        for ctx in matrix_context:
+            args = ctx.injected_args
+            assert len(args) == 4
+            assert ctx.function() == (args['pair_1'], args['pair_2'], args['i'], args['j'])
+
+        assert len(list(filter(lambda x: x.function_name == function_ignore.__name__, context_list))) == 2
+        assert len(list(filter(lambda x: x.ignore, context_list))) == 1
+
+    @staticmethod
+    def _check_injection(context_list, *, versions, global_args=None, pairs=False):
+        if global_args:
+            global_versions = global_args['ignite_versions']
+
+            if isinstance(global_versions, str):
+                assert not pairs
+
+                check_versions = [IgniteVersion(global_versions)]
+            elif isinstance(global_args['ignite_versions'], tuple):
+                assert pairs
+
+                check_versions = [tuple(map(IgniteVersion, global_versions))]
+            elif pairs:
+                check_versions = list(map(lambda x: (IgniteVersion(x[0]), IgniteVersion(x[1])), global_versions))
+            else:
+                check_versions = list(map(IgniteVersion, global_versions))
+        else:
+            check_versions = list(map(IgniteVersion, versions)) if not pairs else \
```

Comment:
What do you think if split the ternary operation on explicit if-else blocks, so every assignment of check_version is explicitly visible? IMHO, it will make code a bit clearer.

Reply:
I disagree.

Reply:
Changed